treemap/parseData.js

```diff
@@ -0,0 +1,125 @@
+        const plotData = [{ name: 'treemap', children: buildTree() }];
+
+        function buildTree() {
+            const treeData = seriesNames.map(d => ({
+                name: d,
+                children: getChuldren(d),
+            }));
+            return treeData;
+        }
+
+        function getChuldren(el) {
+            const children = data.filter(d => d.category === el);
+            return children;
+        }
+
+        return {
+            seriesNames,
+            plotData,
+            data,
+        };
+    });
+}
+
+
+/**
+ * Returns the columns headers from the top of the dataset, excluding specified
+ * @param  {[type]} columns [description]
+ * @return {[type]}         [description]
+ */
+export function getSeriesNames(columns) {
+    const exclude = ['date', 'annotate', 'highlight'];
+    return columns.filter(d => (exclude.indexOf(d) === -1));
+}
+
+/**
+ * Calculates the extent of multiple columns
+ * @param  {[type]} d       [description]
+ * @param  {[type]} columns [description]
+ * @param  {[type]} yMin    [description]
+ * @return {[type]}         [description]
+ */
+export function extentMulti(d, columns, yMin) {
+    const ext = d.reduce((acc, row) => {
+        const values = columns.map(key => row[key])
+            .map((item) => {
+                if (!item || item === '*') {
+                    return yMin;
+                }
+                return Number(item);
+            });
+        const rowExtent = d3.extent(values);
+        if (!acc.max) {
+            acc.max = rowExtent[1];
+            acc.min = rowExtent[0];
+        } else {
+            acc.max = Math.max(acc.max, rowExtent[1]);
+            acc.min = Math.min(acc.min, rowExtent[0]);
+        }
+        return acc;
+    }, {});
+    return [ext.min, ext.max];
+}
+
+/**
+ * Sorts the column information in the dataset into groups according to the column
+ * head, so that the line path can be passed as one object to the drawing function
+ */
+export function getlines(d, group, joinPoints) {
+    const lineData = [];
+    d.forEach((el) => {
+        // console.log(el,i)
+        const column = {};
+        column.name = group;
+        column.date = el.date;
+        column.value = +el[group];
+        column.highlight = el.highlight;
+        column.annotate = el.annotate;
+        if (el[group]) {
+            lineData.push(column);
+        }
+
+        // if(el[group] == false) {
+        //     lineData.push(null)
+        // }
+        if (el[group] === false && joinPoints === false) {
+            lineData.push(null);
+        }
+    });
+    return lineData;
+    // return d.map((el) => {
+    //     if (el[group]) {
+    //         return {
+    //             name: group,
+    //             date: el.date,
+    //             value: +el[group],
+    //             highlight: el.highlight,
+    //             annotate: el.annotate,
+    //         };
+    //     }
+
+    //     return null;
+    // }).filter(i => i);
```

Comment:
If this is archaic, please remove commented code.

Reply:
That whole chunk can go TBH its legacy code from the line chart, deleted